src/transactions/3_castVotes.js

```diff
@@ -18,22 +18,42 @@
  */
 import cryptoModule from '../crypto';
 import { VOTE_FEE } from '../constants';
-import { prepareTransaction, getTimeWithOffset } from './utils';
+import {
+	prepareTransaction,
+	getTimeWithOffset,
+	prependMinusToPublicKeys,
+	prependPlusToPublicKeys,
+	validatePublicKeys,
+} from './utils';
 
 /**
  * @method castVotes
  * @param {Object} Object - Object
  * @param {String} Object.passphrase
- * @param {Array<String>} Object.delegates
+ * @param {Array<String>} Object.votes
+ * @param {Array<String>} Object.unvotes
  * @param {String} Object.secondPassphrase
  * @param {Number} Object.timeOffset
  *
  * @return {Object}
  */
 
-const castVotes = ({ passphrase, delegates, secondPassphrase, timeOffset }) => {
+const castVotes = ({
+	passphrase,
+	votes,
+	unvotes,
+	secondPassphrase,
+	timeOffset,
+}) => {
 	const keys = cryptoModule.getKeys(passphrase);
 
+	const plusPrependedVotes =
+		votes && validatePublicKeys(votes) ? prependPlusToPublicKeys(votes) : [];
```

Comment:
As mentioned in the other review, prefer setting a default value for `votes`/`unvotes` and separating this function into 3 stages:
1. Validation
2. Processing inputs
3. Combining processed inputs